pulp_rpm/app/models.py

```diff
@@ -784,3 +791,254 @@ class ModulemdDefaults(Content):
+                           example, 'Fedora', 'Red Hat Enterprise Linux', or 'Spacewalk'.
+    :ivar variant:         The variant contained in the repository. The current version of the
+                           model is problematic since some old "treeinfo" files don't have
+                           variants, and some new repositories have _several_ variants. Since
+                           we currently use the [general] section of the "treeinfo" file, this
+                           variant will be the first variant (alphabetically) in the distribution
+                           tree.
+    :ivar version:         Equivalent to the "treeinfo" [release] section's 'version' field.
+                           For example, '21', '7.0', and '2.1'
+    :ivar arch:            Equivalent to the "treeinfo" [tree] section's 'arch' field. The
+                           architecture the tree was built for. For example, 'x86_64'.
+    :ivar files:           A somewhat hodgepodge list of files in the distribution. It contains
+                           some of the files described in the "treeinfo" files, and potentially
+                           files described by an XML file called "PULP_DISTRIBUTION.xml".
+    :ivar timestamp:       Equivalent to the "treeinfo" [tree] section's 'build_timestamp' field.
+    :ivar packagedir:      Equivalent to the "treeinfo" [variant-*] section's 'packages' field.
+                           It's possible in the future that distribution trees will have several
+                           different packagedir fields, so we need to fix our model here.
+    :ivar version_sort_index: ?
+
+    """
+
+    TYPE = 'distribution_tree'
+
+    header_version = models.CharField(max_length=10)
+
+    release_name = models.CharField(max_length=50)
+    release_short = models.CharField(max_length=20)
+    release_version = models.CharField(max_length=10)
+    release_is_layered = models.BooleanField(default=False)
+
+    base_product_name = models.CharField(max_length=50, null=True)
+    base_product_short = models.CharField(max_length=20, null=True)
+    base_product_version = models.CharField(max_length=10, null=True)
+
+    # tree
+    arch = models.CharField(max_length=30)
+    build_timestamp = models.FloatField()
+
+    # stage2
+    instimage = models.CharField(max_length=50, null=True)
+    mainimage = models.CharField(max_length=50)
+
+    # media
+    discnum = models.IntegerField()
+    totaldiscs = models.IntegerField()
+
+    class Meta:
+        default_related_name = "%(app_label)s_%(model_name)s"
+        unique_together = (
+            "header_version",
+            "release_name",
+            "release_short",
+            "release_version",
+            "arch",
+            "build_timestamp",
+        )
+
+
+class Checksum(Model):
+    """
+    Distribution Tree Checksum.
+
+    """
+
+    path = models.CharField(max_length=50)
```

Comment:
What is the reasoning behind chosen limits for CharFields? The examples from a few repos in the wild? 
Some of them seem too strict to me. Are we ok to extend the CharField later (if it's needed) with a migration? @bmbouter